src/stores/mainStore.ts

```diff
@@ -103,13 +135,21 @@ export class MainStore {
                     )
                   ) {
                     this.baccarat.giveBankerACard();
+                    timer = setTimeout(() => {
+                      this.baccarat.bankerCards[2].flipped = true;
+                      clearTimeout(timer);
+                    }, 500);
                   }
                 } else if (
                   needThirdCardPlayerRule(this.baccarat.bankerPoints)
                 ) {
                   // player didn't get third card
                   // banker third card according to players rule
                   this.baccarat.giveBankerACard();
+                  const timer = setTimeout(() => {
+                    this.baccarat.playerCards[2].flipped = true;
```

Comment:
What I noticed is that sometimes this value is `undefined`. Long story short, it is safer to write it like this or revalidate game logic.
```typescript
if (this.baccarat.playerCards[2]) {
    this.baccarat.playerCards[2].flipped = true;
}
```